src/scripts/main.js

```diff
@@ -1,3 +1,495 @@
 'use strict';
 
-// write code here
+const body = document.querySelector('body');
+const head = document.querySelector('thead');
+const titles = [...head.children[0].children];
+const tbody = document.querySelector('tbody');
+
+function compareCurrency(cur1, cur2) {
+  const firstCurrency = +cur1.slice(1).split(',').join('');
+  const secCurrency = +cur2.slice(1).split(',').join('');
+
+  if (firstCurrency > secCurrency) {
+    return true;
+  }
+
+  if (firstCurrency < secCurrency) {
+    return false;
+  }
+}
+
+function compareStrings(str1, str2) {
+  const firstRowLowered = str1.toLowerCase();
+  const secondRowLowered = str2.toLowerCase();
+
+  if (firstRowLowered > secondRowLowered) {
+    return true;
+  }
+
+  if (firstRowLowered < secondRowLowered) {
+    return false;
+  }
+}
+
+function sortTable(index) {
+  const table = document.querySelector('tbody');
+  let isSwitching = true;
+  let countSwitch = 0;
+  let sortDirection = 'asc';
+
+  while (isSwitching) {
+    isSwitching = false;
+
+    const rows = table.rows;
+    let shouldRowsSwitch = false;
+    let i;
+
+    for (i = 0; i < rows.length - 1; i++) {
+      shouldRowsSwitch = false;
+
+      const firstRow = rows[i].getElementsByTagName('TD')[index];
+      const secondRow = rows[i + 1].getElementsByTagName('TD')[index];
+
+      const firstRowContent = firstRow.innerHTML;
+      const secondRowContent = secondRow.innerHTML;
+
+      const isFirstRowNum = Number(firstRowContent);
+      const isSecondRowNum = Number(secondRowContent);
+
+      if (sortDirection === 'asc') {
+        if (isFirstRowNum && (isFirstRowNum > isSecondRowNum)) {
+          shouldRowsSwitch = true;
+          break;
+        } else {
+          const isCurrency = firstRowContent.indexOf('$');
+
+          if (isCurrency !== -1) {
+            if (compareCurrency(firstRowContent, secondRowContent)) {
+              shouldRowsSwitch = true;
+              break;
+            };
+          } else {
+            if (compareStrings(firstRowContent, secondRowContent)) {
+              shouldRowsSwitch = true;
+              break;
+            }
+          }
+        }
+      }
+
+      if (sortDirection === 'desc') {
+        if (isFirstRowNum && (isFirstRowNum < isSecondRowNum)) {
+          shouldRowsSwitch = true;
+          break;
+        } else {
+          const isCurrency = firstRowContent.indexOf('$');
+
+          if (isCurrency !== -1) {
+            if (compareCurrency(firstRowContent, secondRowContent) === false) {
+              shouldRowsSwitch = true;
+              break;
+            };
+          } else {
+            if (compareStrings(firstRowContent, secondRowContent) === false) {
+              shouldRowsSwitch = true;
+              break;
+            }
+          }
+        }
+      }
+    }
+
```

Comment:
This almost exaclty dublicates if block above. Consider reorganising your algorithm, changing conditions and their order, maybe creating some helper functions or modifying existing ones and try to find a way to fix this